databroker/cli.py

```diff
@@ -67,6 +85,10 @@ def shape_fix(
         typer.echo(f"Limited to first {limit} BlueskyRuns only")
         items = items[:limit]
 
+    root_map = root_map if root_map is not None else {}
+    root_map = getattr(adapter, "root_map", {}) | root_map
```

Comment:
Should this be `or` instead of `|`?

```python
>>> None or 1
1
>>> None | 1
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for |: 'NoneType' and 'int'
```

Reply:
That should work as an [in-place `update` operation (or union)](https://peps.python.org/pep-0584/). The first line ensures you're working with dicts.

Reply:
Oh right, not used to that syntax yet. Cool.